addon.py

```diff
@@ -223,17 +211,47 @@ def genre_list(self):
                 handle=self.addon_handle, url=url, listitem=li, isFolder=True)
 
         xbmcplugin.endOfDirectory(self.addon_handle)
+        
+    def newest_albums_list(self):
+        """
+        Display newest album list.
+        """
+
+        size = self.album_list_size
+
+        xbmcplugin.setContent(self.addon_handle, "albums")
+
+        for album in self.connection.walk_album_list('newest',size,None,None,None):
+            self.add_album(album, show_artist=True)
+
+        xbmcplugin.endOfDirectory(self.addon_handle)
+        
+    def albums_by_genre_list(self):
+        """
+        Display album list by genre menu.
+        """
+
+        size = self.album_list_size
+        genre = self.addon_args["foldername"][0].decode("utf-8")
+
+        xbmcplugin.setContent(self.addon_handle, "albums")
+
+        for album in self.connection.walk_album_list('byGenre',size,None,None,genre):
+            self.add_album(album, show_artist=True)
+
+        xbmcplugin.endOfDirectory(self.addon_handle)
 
     def albums_by_genre_list(self):
         """
         Display album list by genre menu.
         """
 
+        size = self.album_list_size
         genre = self.addon_args["foldername"][0].decode("utf-8")
 
         xbmcplugin.setContent(self.addon_handle, "albums")
 
-        for album in self.connection.walk_album_list_genre(genre):
+        for album in self.connection.walk_album_list('byGenre',size,None,None,genre):
```

Comment:
I think you can also write `self.connection.walk_album_list('newest', size, genre=genre)`. The `None` are optional kwargs (they have defaults).